lib/phoenix/live_dashboard/live/page_live.ex

```diff
@@ -39,38 +57,126 @@ defmodule Phoenix.LiveDashboard.PageLive do
     end
   end
 
   def mount(_params, _session, socket) do
-    {:ok, push_redirect(socket, to: live_dashboard_path(socket, :home, node(), []))}
+    {:ok, redirect_to_current_node(socket)}
+  end
+
+  defp assign_mount(socket, page_live, page_session, params, session) do
+    socket =
+      Phoenix.LiveView.assign(socket, :page, %__MODULE__{
+        page_live: page_live,
+        session: page_session
+      })
+
+    with %Socket{redirected: nil} = socket <- assign_params(socket, params),
+         %Socket{redirected: nil} = socket <- assign_node(socket, params),
+         %Socket{redirected: nil} = socket <- assign_refresh(socket),
+         %Socket{redirected: nil} = socket <- assign_capabilities(socket, session) do
+      maybe_apply_module(socket, :mount, [params, page_session], &{:ok, &1})
+    else
+      %Socket{} = redirected_socket -> {:ok, redirected_socket}
+    end
+  end
+
+  defp assign_params(socket, params) do
+    update_page(socket, params: params, info: info(params), route: route(params))
+  end
+
+  defp route(%{"page" => page}), do: String.to_existing_atom(page)
+
+  defp info(%{"info" => info} = params), do: {info, Map.delete(params, "info")}
+  defp info(%{}), do: nil
+
+  defp assign_node(socket, params) do
+    param_node = Map.fetch!(params, "node")
+
+    if found_node = Enum.find(nodes(), &(Atom.to_string(&1) == param_node)) do
+      if connected?(socket) do
+        :net_kernel.monitor_nodes(true, node_type: :all)
+      end
+
+      update_page(socket, node: found_node, nodes: nodes())
+    else
+      redirect_to_current_node(socket)
+    end
+  end
+
+  def assign_refresh(socket) do
+    page_live = socket.assigns.page.page_live
+
+    socket
+    |> update_page(
+      refresher?: page_live.__page_live__(:refresher?),
+      refresh: page_live.__page_live__(:refresh),
+      refresh_options: page_live.__page_live__(:refresh_options)
+    )
+    |> init_schedule_refresh()
+  end
+
+  defp init_schedule_refresh(socket) do
+    if connected?(socket) and socket.assigns.page.refresher? do
+      schedule_refresh(socket)
+    else
+      assign(socket, timer: nil)
+    end
+  end
+
+  defp schedule_refresh(socket) do
+    assign(socket, timer: Process.send_after(self(), :refresh, socket.assigns.page.refresh * 1000))
+  end
+
+  def assign_capabilities(socket, session) do
+    capabilities = Phoenix.LiveDashboard.SystemInfo.ensure_loaded(socket.assigns.page.node)
+
+    update_page(socket,
+      metrics: capabilities.dashboard && session["metrics"],
+      os_mon: capabilities.os_mon,
+      request_logger: capabilities.dashboard && session["request_logger"],
+      dashboard_running?: capabilities.dashboard
+    )
   end
 
   defp maybe_apply_module(socket, fun, params, default) do
-    if function_exported?(socket.assigns.module, fun, length(params) + 1) do
-      apply(socket.assigns.module, fun, params ++ [socket])
+    if function_exported?(socket.assigns.page.page_live, fun, length(params) + 1) do
```

Comment:
Maybe do you still prefer `module`?

Reply:
I personally prefer it, yeah. Or `page_module`.